pkg/securitypolicy/framework.rego

```diff
@@ -45,6 +45,13 @@ layerPaths_ok(layers) {
     }
 }
 
+overlay_dirs_encrypted() {
+    some encrypted_path in data.metadata.encrypted_devices {
+        startswith(input.upperDir, encrypted_path)
```

Comment:
Both of these conditions cannot be true simultaneously. Are you testing to see if either is true? As written, this will never reach the second check (i.e. the first check being false will short-circuit the test).

If what you want is to check that there exists a path which either starts with the upperDir or workDir of this overlay, then what you want is the following code:

``` rego
path_encrypted(path) {
    startswith(input.upperDir, path)
}

path_encrypted(path) {
    startswith(input.workDir, path)
}

default overlay_dirs_encrypted := false
overlay_dirs_encrypted {
    some path in data.metadata.encrypted_devices {
       path_encrypted(path)
    }
}
```

This code creates a rule function named `path_encrypted` which takes a path and has two possible predicates where it is true (i.e. the two different prefixes from the input). You then have a rule, `overlay_dirs_encrypted`, which has a default value of false (important) and then does the existence check. I think this is probably the logic you want.

Reply:
this was removed.